### DISCUSSION TITLE
Introducing a notch to my geometry - Neper

### DISCUSSION THREAD

**Discussion by Sofia-Y** [neperfepx/neper#107] [Q&A] [answered]
Dear all,

I am glad I found this platform for discussions and questions. I would like to know if it is possible to introduce a notch to my geometry in Neper; (e.g., how can I use Neper to build a plate with a notch/ deep notch? ).

Thanks.

**Comment by rquey** [comment, marked as answer]
Hi @Sofia-Y - Here is an example of what can be done:
```
$ neper -T -n 1000 -dim 2 -transform "cut(ecylinder(0,0.5,0,0,0,1,1,0,0,0,1,0,0.5,0.05))"
$ neper -V n1000-id1.tess -cameracoo x:y:10 -cameralookat x:y:z -datacellcol id -print img
```
![img](https://user-images.githubusercontent.com/2659376/107570569-01b39e00-6bea-11eb-9df2-edd16dedbdad.png)

**Reply by Sofia-Y** [reply]
Hi, @rquey !
I appreciate your assistance. My aim is to get an input file for ABAQUS of a plate with a notch. I am currently running few examples to get used to the coordinates of the notch. However, I noticed that when I cut my model I get an error when using the -M module. The mesh would fail in some faces. If you try to run this same example you have given above and get the input file you will get the following error: How can I avoid this and get my input file ?
```
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.1.2
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt
Info   : Running on 32 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2020, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/sy422/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n1000-id1.tess -elttype tri -format inp
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `n1000-id1.tess'...
Info   :     [i] Parsed file `n1000-id1.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.007901)
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing...  18% (0|0.9/99%| 1%| 0%)Error  :     > Meshing of face 80 failed
Aborted
```

**Reply by rquey** [reply]
Correct. The `-transform cut` feature is experimental, and tessellation generation or meshing do not always succeed. Try to use a different `-id` value for the tessellation, e.g.

```
$ neper -T -n 1000 -dim 2 -transform "cut(ecylinder(0,0.5,0,0,0,1,1,0,0,0,1,0,0.5,0.05))" -id 3
```

Also, I did not enable regularization in my example, but this is something you want to do; so, add `-reg 1`:

```
$ neper -T -n 1000 -dim 2 -transform "cut(ecylinder(0,0.5,0,0,0,1,1,0,0,0,1,0,0.5,0.05))" -id 3 -reg 1
```

**Reply by ghost** [reply]
Hello, I've tried to use the transform cut ecylindre command : 
` neper -T -n 200 -dim 2 -transform "cut(ecylinder(0,0.5,0,0,0,1,1,0,0,0,1,0,0.5,0.001))" -id 3`
 to have something that looks like this : 
![image](https://github.com/user-attachments/assets/1541afe5-801d-4140-a050-b755f39aa2d2)
but I keep getting holes in the mesh and not a line as : 
![image](https://github.com/user-attachments/assets/4e771c8b-533f-4225-a011-76d485e6f707)
Is there a way to remediate to this? Thanks !

**Reply by rquey** [reply]
For now, not really. The cutting algorithm works well only when the local curvature is of the order of the grain size or larger.  It is clearly not the case here.

**Reply by ghost** [reply]
Ok thank you

**Reply by sudeepkolati** [reply]
How to get connectivity information of the grains for the above notch

**Reply by sudeepkolati** [reply]
and also how to keep finer grains in the vicinity of notch

**Reply by sudeepkolati** [reply]
![voronoi_sen](https://github.com/user-attachments/assets/2998453d-7e15-4c8d-9cda-d00b71a0e205)
how to generate a mesh like this
